src/Data/List/Relation/Unary/Any/Properties.agda

```diff
@@ -459,6 +467,65 @@ module _ {P : A → Set p} where
   tabulate⁻ {suc n} (here p)   = fzero , p
   tabulate⁻ {suc n} (there p) = Prod.map fsuc id (tabulate⁻ p)
 
+------------------------------------------------------------------------
+-- filter
+
+module _ {P : A → Set p} {Q : A → Set q} (Q? : U.Decidable Q) where
+
+  filter⁺ : ∀ {xs} → (p : Any P xs) → Any P (filter Q? xs) ⊎ ¬ Q (Any.lookup p)
+  filter⁺ {x ∷ xs} (here px) with Q? x
+  ... | true  because _       = inj₁ (here px)
+  ... | false because ofⁿ ¬Qx = inj₂ ¬Qx
+  filter⁺ {x ∷ xs} (there p) with Q? x
+  ... | true  because _       = Sum.map₁ there (filter⁺ p)
+  ... | false because _       = filter⁺ p
+
+  filter⁻ : ∀ {xs} → Any P (filter Q? xs) → Any P xs
+  filter⁻ {x ∷ xs} p with Q? x
+  filter⁻ {x ∷ xs} (here px) | true  because _ = here px
```

Comment:
I believe using `with does (Q? x)` would allow you  to leave the `because _` out.